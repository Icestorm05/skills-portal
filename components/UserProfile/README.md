# User Profile

## Functionality

The user profile component displays all of a user's profile details. Also, allows a user to edit their biography.

## Usage

### Props

`<user-profile>` supports the following custom component attributes:

| attribute | type | description
| --- | --- | ---
| `disabled` | Boolean *optional* | Whether or not the buttons are disabled or not. Defaults to `false`.
| `editable` | Boolean *optional* | Whether the biography can be edited or not. Defaults to `false`.
| `profile` | Object | The user profile.

### Events

`<user-profile>` emits the following events:
| event | type | description
| --- | --- | ---
| `updated-biography` | String | The updated biography.

### Schemas

The `profile` prop requires the following to exist within the object:

| attribute | type | description
| --- | --- | ---
| `Biography` | String | The biography of the profile.
| `Company` | String | The company the owner of the profile works for.
| `Director` | String | The director of the owner of the profile.
| `EmailAddress` | String | The email address of the owner of the profile.
| `FirstLineReporting` | String | The person who is the first line reporting for the owner of the profile.
| `FullName` | String | The full name of the person who owns the profile.
| `JobTitle` | String | The job title of the person who owns the profile.
| `SecondLineReporting` | String | The person who is the second line reporting for the owner of the profile.

### Example

```html
<user-profile profile="profile"
              readonly
              @updated-biography="updateBio"/>
```

```ts
import Component from 'nuxt-class-component';
import Vue from 'vue';

import UserProfile from '~/components/UserProfile/UserProfile.vue';

@Component({
    components: {
        UserProfile,
    },
})
export default class GenericComponent extends Vue {
    private profile = {
        Biography: 'This is a test biography',
        Company: 'BAS Perm',
        Director: 'Janet Doe',
        EmailAddress: 'John.Doe@test.com',
        FirstLineReporting: 'Jake Doe',
        FullName: 'John Doe',
        JobTitle: 'Innovation Developer',
        SecondLineReporting: 'Jane Doe',
    }

    private updateBio(biography: string) {
        console.log(biography);
    }
}
```