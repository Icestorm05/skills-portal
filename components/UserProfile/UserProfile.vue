<template lang="pug">
    v-card(raised)#UserProfile.elevation-12
            v-layout(row
                     wrap)
                v-flex(sm4 md3).primary#UserProfileLeft
                    v-card-title(primary-title).fill-height
                        v-layout(column
                                 align-center
                                 justify-center
                                 fill-width).text-xs-center
                            v-avatar(size="150")#UserProfileAvatar
                                avatar-image(:image-name="profile.MainStaffNumber"
                                             :renderAsBase64="true")
                            h3.headline.white--text.text-xs-center#UserProfileFullName {{ profile.FullName }}
                            i.subheading.warning--text#UserProfileJobTitle.fill-width {{ profile.JobTitle }}
                            span.white--text#UserProfileCompany.fill-width {{ profile.Company }}
                v-flex(sm8 md9)#UserProfileRight
                    v-card-text.fill-height
                        v-form(@submit.prevent="save").fill-height.layout.column#UserProfileForm
                            v-flex.white-space
                                v-text-field(v-if="editable"
                                            label="Biography"
                                            clearable
                                            color="primary"
                                            counter="100"
                                            placeholder="Enter your Biography here"
                                            multi-line
                                            name="biography"
                                            v-model="biography"
                                            data-vv-name="biography"
                                            v-validate="'required|max:100'"
                                            :error-messages="errors.collect('biography')"
                                            :disabled="disabled")#UserProfileBiography
                                template(v-else)
                                    b.pa-0.primary--text Biography
                                    p.light.mb-0#UserProfileBiography {{ biography }}
                            v-divider.no-flex.my-3
                            p.light.mb-1
                                b.primary--text.mr-1 Email Address:
                                span#UserProfileEmailAddress.break-word {{ profile.EmailAddress }}
                            p.light.mb-1
                                b.primary--text.mr-1 Director:
                                span#UserProfileDirector.break-word {{ profile.Director }}
                            p.light.mb-1
                                b.primary--text.mr-1 First Line Reporting:
                                span#UserProfileFirstLineReporting.break-word {{ profile.FirstLineReporting }}
                            p.light.mb-1
                                b.primary--text.mr-1 Second Line Reporting:
                                span#UserProfileSecondLineReporting.break-word {{ profile.SecondLineReporting }}
                            v-divider(v-if="editable").no-flex.my-3
                            v-card-actions(v-if="editable").fill-width.pa-0
                                v-layout(full-width
                                        justify-center)
                                    v-btn(color="primary"
                                          flat
                                          aria-label="Save Profile"
                                          type="submit"
                                          :disabled="disabled")#UserProfileSaveButton Save
</template>

<script lang="ts">
import Component from "nuxt-class-component";
import Vue from "vue";
import { Prop, Watch } from "vue-property-decorator";

import { IProfile } from "types/profile";

import VAvatar from "vuetify/es5/components/VAvatar";
import VBtn from "vuetify/es5/components/VBtn";
import VCard, {
  VCardActions,
  VCardText,
  VCardTitle
} from "vuetify/es5/components/VCard";
import VDivider from "vuetify/es5/components/VDivider";
import VForm from "vuetify/es5/components/VForm";
import { VFlex, VLayout } from "vuetify/es5/components/VGrid";
import VTextField from "vuetify/es5/components/VTextField";
import AvatarImage from "~/components/AvatarImage/AvatarImage.vue";

@Component({
  components: {
    AvatarImage,
    VAvatar,
    VBtn,
    VCard,
    VCardActions,
    VCardText,
    VCardTitle,
    VDivider,
    VFlex,
    VForm,
    VLayout,
    VTextField
  },
  $_veeValidate: {
    validator: "new"
  }
})
export default class UserProfile extends Vue {
  @Prop({ type: Boolean, default: false })
  private disabled: boolean;
  @Prop({ type: Boolean, default: false })
  private editable: boolean;
  @Prop({ type: Object, required: true })
  private profile: IProfile;

  private biography = "";

  @Watch("profile", { deep: true, immediate: true })
  private profileChanged() {
    this.biography = this.profile.Biography.replace(/\\n/g, "\n");
  }

  private async save() {
    if (await this.$validator.validateAll()) {
      this.$emit("updated-biography", this.biography);
    }
  }
}
</script>

<style lang="stylus" scoped>
.no-flex {
    flex: none;
}

.light {
    color: rgba(0, 0, 0, 0.54);
}

.fill-width {
    width: 100%;
}

#UserProfileLeft {
    flex-basis: 100%;
}

#UserProfile /deep/ .input-group__error {
    flex: none;
}

.white-space {
    white-space: pre-wrap;
}

.break-word {
    word-break: break-word;
}
</style>
