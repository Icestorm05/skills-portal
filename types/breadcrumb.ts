import { Location } from 'vue-router';

export interface IBreadcrumb {
    link: Location;
    text: string;
}