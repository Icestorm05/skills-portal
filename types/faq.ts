export interface IFaq {
    question: string;
    answers: IFaqAnswer[];
}

export interface IFaqAnswer {
    type: 'p' | 'b' | 'ul' | 'table';
    text: string | string[] | string[][];
}