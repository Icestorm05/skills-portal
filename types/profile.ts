export interface IProfile {
    Biography: string;
    Company: string;
    Director: string;
    EmailAddress: string;
    EmployeeId: number;
    FirstLineReporting: string;
    FullName: string;
    JobTitle: string;
    KnownAs: string;
    MainStaffNumber: string;
    SecondLineReporting: string;
    Surname: string;
    children: IProfile[];
}