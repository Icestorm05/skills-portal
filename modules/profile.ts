import { IProfile } from "types/profile";

/**
 * Calculates how deep a profile is in a hierarchy of profiles.
 * @param profiles {IProfile[]} The hierarchy of profiles.
 * @param profileToFind {IProfile} The profile to find.
 * @param level {number?} The depth level when no profile is found. Defaults to 0.
 * @return {number} The depth level.
 */
export const deepLevel = (
  profiles: IProfile[],
  profileToFind: IProfile,
  level: number = 0
): number => {
  return profiles.reduce((depth, profile: IProfile) => {
    if (profile.EmployeeId === profileToFind.EmployeeId) {
      return level + 1;
    } else if (profile.children) {
      const deepDepth = deepLevel(profile.children, profileToFind, level + 1);
      if (deepDepth) {
        return deepDepth;
      }
    }
    return depth;
  }, 0);
};

/**
 * Searches a hierarchy of profiles for one that matches a specified employee ID.
 * @param profiles {IProfile[]} The profile hierarchy.
 * @param id {number} The employee ID to find.
 * @return {IProfile | null} The matched profile, or null if not found.
 */
export const findProfile = (
  profiles: IProfile[],
  id: number
): IProfile | null => {
  return profiles.reduce((matchedProfile: IProfile | null, currProfile) => {
    if (!matchedProfile) {
      if (currProfile.EmployeeId === id) {
        return currProfile;
      } else if (currProfile.children) {
        return findProfile(currProfile.children, id);
      } else {
        return null;
      }
    }
    return matchedProfile;
  }, null);
};
