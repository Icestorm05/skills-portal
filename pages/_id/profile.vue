<template lang="pug">
    #ProfilePage
        user-profile(:profile="profile"
                     :disabled="inProgress"
                     :editable="currentDepth === 1"
                     @updated-biography="updateBiography")
</template>

<script lang="ts">
import Component from "nuxt-class-component";
import Vue from "vue";
import { Getter, State } from "vuex-class";

import { IProfile } from "types/profile";

import UserProfile from "~/components/UserProfile/UserProfile.vue";

@Component({
  components: {
    UserProfile
  }
})
export default class ProfilePage extends Vue {
  @Getter("profile/current") private profile: IProfile;
  @Getter("profile/currentDepth") private currentDepth: number;

  @Getter("progress/isVisible") private inProgress: boolean;

  private async updateBiography(Biography: string) {
    try {
      this.$store.commit("progress/show");
      await this.$store.dispatch("profile/edit", {
        ...this.profile,
        Biography
      });
      this.$store.commit("snackbar/show", {
        message: "Successfully updated bio.",
        color: "success"
      });
    } catch (err) {
      this.$store.commit("snackbar/show", {
        message: "Failed to update bio.",
        color: "error"
      });
    } finally {
      this.$store.commit("progress/hide");
    }
  }
}
</script>
