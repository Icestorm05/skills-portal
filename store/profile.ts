import { IProfile } from "types/profile";
import { deepLevel, findProfile } from "~/modules/profile";

const defaultProfile = {
  Biography: "",
  Director: "",
  EmailAddress: "",
  EmployeeId: 0,
  FirstLineReporting: "",
  FullName: "",
  JobTitle: "",
  KnownAs: "",
  SecondLineReporting: "",
  Surname: "",
  children: []
};

export const state = () => ({ ...defaultProfile });

export const mutations = {
  set(currentProfile: IProfile, profile: IProfile) {
    Object.keys(currentProfile).forEach(
      key => (currentProfile[key] = profile[key])
    );
  },
  updateBiography(currentProfile: IProfile, biography: string) {
    currentProfile.Biography = biography;
  },
  reset(currentProfile: IProfile) {
    Object.keys(currentProfile).forEach(
      key => (currentProfile[key] = defaultProfile[key])
    );
  }
};

export const actions = {
  async get({ commit }) {
    const data = await this.$axios.$get("/profile");
    commit("set", data.profile);
  },
  async edit({ commit }, profile: IProfile) {
    const url = `/employees/${profile.EmployeeId}`;
    await this.$axios.$put(url, { employee: profile });
    commit("updateBiography", profile.Biography);
  }
};

export const getters = {
  current(currentProfile: IProfile, localGetters, globalState) {
    const id = globalState.route
      ? parseInt(globalState.route.params.id, 10)
      : 0;
    const profile = findProfile(
      [currentProfile],
      id || currentProfile.EmployeeId
    );
    return profile ? profile : currentProfile;
  },
  currentDepth(currentProfile: IProfile, localGetters) {
    return localGetters.current.EmployeeId
      ? deepLevel([currentProfile], localGetters.current)
      : 0;
  },
  hasTeam(currentProfile: IProfile, localGetters) {
    return localGetters.current.children.length ? true : false;
  }
};
