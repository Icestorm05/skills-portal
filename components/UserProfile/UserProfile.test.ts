import UserProfile from "./UserProfile.vue";
import flushPromises from "flush-promises";
import VeeValidate from "vee-validate";
import Vue from "vue";
import { createLocalVue, mount } from "@vue/test-utils";

import { IProfile } from "types/profile";

declare const expect: jest.Expect;

const profile: IProfile = {
  Biography: "Test Biography",
  Company: "Test Company",
  Director: "John Doe",
  EmailAddress: "test@test.com",
  EmployeeId: 1,
  FirstLineReporting: "Jane Doe",
  FullName: "Jake Doe",
  JobTitle: "Innovation Developer",
  KnownAs: "Jake",
  MainStaffNumber: "123",
  SecondLineReporting: "Julie Doe",
  Surname: "Doe",
  children: []
};
const profileWithBlankBio = { ...profile, Biography: "" };
const profileWithLongBio = {
  ...profile,
  Biography: `Very long bio. Very long bio. Very long bio.
Very long bio. Very long bio. Very long bio. Very long bio. Very long bio. Very long bio.
Very long bio. Very long bio. Very long bio. Very long bio. Very long bio. Very long bio.
Very long bio. Very long bio. Very long bio. Very long bio. Very long bio. Very long bio.`
};

const localVue = createLocalVue();
localVue.use(VeeValidate);

describe("User Profile", () => {
  test("is a Vue instance", () => {
    const wrapper = mount(UserProfile, { localVue, propsData: { profile } });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("snapshot has not changed", () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: true, profile }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("readonly snapshot has not changed", () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: false, profile }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("textarea is displayed when editable is set to true", () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: true, profile }
    });
    expect(wrapper.find("textarea").exists()).toBeTruthy();
  });

  test("textarea is not displayed when editable is set to false", () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: false, profile }
    });
    expect(wrapper.find("textarea").exists()).toBeFalsy();
  });

  test("save button is displayed when editable is set to true", () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: true, profile }
    });
    expect(wrapper.find("#UserProfileSaveButton").text()).toEqual("Save");
  });

  test("save button is not displayed when editable is set to false", () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: false, profile }
    });
    expect(wrapper.find("#UserProfileSaveButton").exists()).toBeFalsy();
  });

  test("left side of profile card displays avatar", async () => {
    const wrapper = mount(UserProfile, { localVue, propsData: { profile } });
    await wrapper.vm.$nextTick();
    await wrapper.vm.$nextTick();
    const elem = wrapper.find("#UserProfileLeft #UserProfileAvatar > img");
    expect(elem.exists()).toBeTruthy();
    expect(elem.attributes().src).toEqual("/765-default-avatar.jpg");
  });

  test("left side of profile card displays full name", () => {
    const wrapper = mount(UserProfile, { localVue, propsData: { profile } });
    const elem = wrapper.find("#UserProfileLeft #UserProfileFullName");
    expect(elem.text()).toEqual(profile.FullName);
  });

  test("left side of profile card displays job title", () => {
    const wrapper = mount(UserProfile, { localVue, propsData: { profile } });
    const elem = wrapper.find("#UserProfileLeft #UserProfileJobTitle");
    expect(elem.text()).toEqual(profile.JobTitle);
  });

  test("left side of profile card displays company", () => {
    const wrapper = mount(UserProfile, { localVue, propsData: { profile } });
    const elem = wrapper.find("#UserProfileLeft #UserProfileCompany");
    expect(elem.text()).toEqual(profile.Company);
  });

  test("right side of profile card displays biography in readonly mode", () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: false, profile }
    });
    const elem = wrapper.find("#UserProfileRight #UserProfileBiography");
    expect(elem.text()).toEqual(profile.Biography);
  });

  test("right side of profile card displays biography in editable mode", () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: true, profile }
    });
    const elem = wrapper.find("#UserProfileRight #UserProfileBiography")
      .element as HTMLTextAreaElement;
    expect(elem.value).toEqual(profile.Biography);
  });

  test("right side of profile card displays email address", () => {
    const wrapper = mount(UserProfile, { localVue, propsData: { profile } });
    const elem = wrapper.find("#UserProfileRight #UserProfileEmailAddress");
    expect(elem.text()).toEqual(profile.EmailAddress);
  });

  test("right side of profile card displays director", () => {
    const wrapper = mount(UserProfile, { localVue, propsData: { profile } });
    const elem = wrapper.find("#UserProfileRight #UserProfileDirector");
    expect(elem.text()).toEqual(profile.Director);
  });

  test("right side of profile card displays first line reporting", () => {
    const wrapper = mount(UserProfile, { localVue, propsData: { profile } });
    const elem = wrapper.find(
      "#UserProfileRight #UserProfileFirstLineReporting"
    );
    expect(elem.text()).toEqual(profile.FirstLineReporting);
  });

  test("right side of profile card displays second line reporting", () => {
    const wrapper = mount(UserProfile, { localVue, propsData: { profile } });
    const elem = wrapper.find(
      "#UserProfileRight #UserProfileSecondLineReporting"
    );
    expect(elem.text()).toEqual(profile.SecondLineReporting);
  });

  test("biography counter increments and decrements", () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: true, profile: profileWithBlankBio }
    });
    const counter = wrapper.find("#UserProfileRight .input-group__counter");
    expect(counter.text()).toEqual("0 / 100");

    wrapper.setProps({ profile });
    expect(counter.text()).toEqual(`${profile.Biography.length} / 100`);

    wrapper.setProps({ profile: profileWithBlankBio });
    expect(counter.text()).toEqual("0 / 100");
  });

  test("clicking the save button fires the save function", () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: true, profile }
    });

    const spy = jest.fn();
    wrapper.setData({ save: spy });

    wrapper.find("#UserProfileForm").trigger("submit");
    expect(spy).toHaveBeenCalled();
  });

  test("clicking the save button emits the updated biography event", async () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: true, profile }
    });

    expect(wrapper.emitted("updated-biography")).toBeFalsy();
    wrapper.find("#UserProfileForm").trigger("submit");
    await flushPromises();

    expect(wrapper.emitted("updated-biography")).toBeTruthy();
    expect(wrapper.emitted("updated-biography")[0]).toEqual([
      profile.Biography
    ]);
  });

  test("biography required error appears when attempting to save with a blank bio", async () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: true, profile: profileWithBlankBio }
    });

    let error = wrapper.find(".input-group__error");
    expect(error.exists()).toBeFalsy();
    expect(wrapper.vm.$validator.errors.count()).toEqual(0);

    wrapper.find("#UserProfileForm").trigger("submit");
    await Vue.nextTick();

    error = wrapper.find(".input-group__error");
    expect(error.text()).toEqual("The biography field is required.");
    expect(wrapper.vm.$validator.errors.count()).toEqual(1);
  });

  test("biography counter error appears with a bio greater than 100 characters long", async () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: true, profile: profileWithLongBio }
    });

    let error = wrapper.find(".input-group__error");
    expect(error.exists()).toBeFalsy();
    expect(wrapper.vm.$validator.errors.count()).toEqual(0);

    wrapper.find("#UserProfileForm").trigger("submit");
    await Vue.nextTick();

    error = wrapper.find(".input-group__error");
    expect(error.text()).toEqual(
      "The biography field may not be greater than 100 characters."
    );
    expect(wrapper.vm.$validator.errors.count()).toEqual(1);
  });

  test("updating the profile prop updates the biography data", () => {
    const wrapper = mount(UserProfile, {
      localVue,
      propsData: { editable: true, profile }
    });
    expect(wrapper.vm.$data.biography).toEqual(profile.Biography);
    wrapper.setProps({ profile: profileWithBlankBio });
    expect(wrapper.vm.$data.biography).toEqual(profileWithBlankBio.Biography);
  });
});
