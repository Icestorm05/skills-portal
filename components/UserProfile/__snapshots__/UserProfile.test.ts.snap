// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Profile readonly snapshot has not changed 1`] = `
<div class="elevation-12 card card--raised" style="height: auto;" id="UserProfile">
    <div class="layout row wrap">
        <div class="flex primary sm4 md3" id="UserProfileLeft">
            <div class="card__title fill-height card__title--primary">
                <div class="layout text-xs-center column align-center justify-center fill-width">
                    <div id="UserProfileAvatar" class="avatar" style="height: 150px; width: 150px;">
                        <!---->
                    </div>
                    <h3 id="UserProfileFullName" class="headline white--text text-xs-center">Jake Doe</h3>
                    <i id="UserProfileJobTitle" class="subheading warning--text fill-width">Innovation Developer</i><span id="UserProfileCompany" class="white--text fill-width">Test Company</span></div>
            </div>
        </div>
        <div class="flex sm8 md9" id="UserProfileRight">
            <div class="card__text fill-height">
                <form novalidate="novalidate" id="UserProfileForm" class="fill-height layout column">
                    <div class="flex white-space">
                        <b class="pa-0 primary--text">Biography</b>
                        <p id="UserProfileBiography" class="light mb-0">Test Biography</p>
                    </div>
                    <hr class="divider no-flex my-3">
                    <p class="light mb-1">
                        <b class="primary--text mr-1">Email Address:</b><span id="UserProfileEmailAddress" class="break-word">test@test.com</span></p>
                    <p class="light mb-1">
                        <b class="primary--text mr-1">Director:</b><span id="UserProfileDirector" class="break-word">John Doe</span></p>
                    <p class="light mb-1">
                        <b class="primary--text mr-1">First Line Reporting:</b><span id="UserProfileFirstLineReporting" class="break-word">Jane Doe</span></p>
                    <p class="light mb-1">
                        <b class="primary--text mr-1">Second Line Reporting:</b><span id="UserProfileSecondLineReporting" class="break-word">Julie Doe</span></p>
                    <!---->
                    <!---->
                </form>
            </div>
        </div>
    </div>
</div>
`;

exports[`User Profile snapshot has not changed 1`] = `
<div class="elevation-12 card card--raised" style="height: auto;" id="UserProfile">
    <div class="layout row wrap">
        <div class="flex primary sm4 md3" id="UserProfileLeft">
            <div class="card__title fill-height card__title--primary">
                <div class="layout text-xs-center column align-center justify-center fill-width">
                    <div id="UserProfileAvatar" class="avatar" style="height: 150px; width: 150px;">
                        <!---->
                    </div>
                    <h3 id="UserProfileFullName" class="headline white--text text-xs-center">Jake Doe</h3>
                    <i id="UserProfileJobTitle" class="subheading warning--text fill-width">Innovation Developer</i><span id="UserProfileCompany" class="white--text fill-width">Test Company</span></div>
            </div>
        </div>
        <div class="flex sm8 md9" id="UserProfileRight">
            <div class="card__text fill-height">
                <form novalidate="novalidate" id="UserProfileForm" class="fill-height layout column">
                    <div class="flex white-space">
                        <div class="input-group input-group--dirty input-group--placeholder input-group--text-field input-group--multi-line primary--text" data-vv-id="1" aria-required="true" aria-invalid="false">
                            <label for="UserProfileBiography">Biography</label>
                            <div class="input-group__input">
                                <textarea name="biography" data-vv-name="biography" id="UserProfileBiography" tabindex="0" placeholder="Enter your Biography here" rows="5"></textarea>
                                <i aria-hidden="true" class="icon material-icons input-group__append-icon input-group__icon-cb input-group__icon-clearable">clear</i>
                            </div>
                            <div class="input-group__details">
                                <!---->
                                <div class="input-group__counter">14 / 100</div>
                            </div>
                        </div>
                    </div>
                    <hr class="divider no-flex my-3">
                    <p class="light mb-1">
                        <b class="primary--text mr-1">Email Address:</b><span id="UserProfileEmailAddress" class="break-word">test@test.com</span></p>
                    <p class="light mb-1">
                        <b class="primary--text mr-1">Director:</b><span id="UserProfileDirector" class="break-word">John Doe</span></p>
                    <p class="light mb-1">
                        <b class="primary--text mr-1">First Line Reporting:</b><span id="UserProfileFirstLineReporting" class="break-word">Jane Doe</span></p>
                    <p class="light mb-1">
                        <b class="primary--text mr-1">Second Line Reporting:</b><span id="UserProfileSecondLineReporting" class="break-word">Julie Doe</span></p>
                    <hr class="divider no-flex my-3">
                    <div class="card__actions fill-width pa-0">
                        <div class="layout full-width justify-center">
                            <button type="submit" class="btn btn--flat primary--text" aria-label="Save Profile" id="UserProfileSaveButton">
                                <div class="btn__content">Save</div>
                            </button>
                        </div>
                    </div>
                </form>
            </div>
        </div>
    </div>
</div>
`;
